src/main/java/uk/nhs/digital/nhsconnect/nhais/service/edifact_to_fhir/PatchTransactionMapper.java

```diff
@@ -2,7 +2,35 @@
 
 import uk.nhs.digital.nhsconnect.nhais.model.edifact.Transaction;
 import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatch;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatchOperation;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatchRemoval;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentValue;
 
 public interface PatchTransactionMapper {
     AmendmentPatch map(Transaction transaction);
+
+    String REMOVE_INDICATOR = "%";
+
+    default AmendmentPatch createAmendmentPatch(String element, String path) {
+        AmendmentPatch amendmentPatch = new AmendmentPatch();
+        if (element == null) {
+            return null;
+        }
+        if (element.isBlank()) {
+            amendmentPatch.setOp(AmendmentPatchOperation.REPLACE);
+            amendmentPatch.setPath(path);
+            amendmentPatch.setValue(null);
+        } else if (element.equals(REMOVE_INDICATOR)) {
+            AmendmentPatchRemoval amendmentRemoval = new AmendmentPatchRemoval();
+            amendmentRemoval.setOp(AmendmentPatchOperation.REMOVE);
+            amendmentRemoval.setPath(path);
+            return amendmentRemoval;
+        } else {
+            amendmentPatch.setOp(AmendmentPatchOperation.REPLACE);
+            amendmentPatch.setPath(path);
+            amendmentPatch.setValue(AmendmentValue.from(element));
+        }
+
+        return amendmentPatch;
+    }
```

Comment:
`AmendmentPatch` has all args constructor. Why don't use that?
````java
default AmendmentPatch createAmendmentPatch(String value, String path) {
    if (value == null) {
        return null;
    }
    if (value.isBlank()) {
        return new AmendmentPatch(AmendmentPatchOperation.REPLACE, path, null);
    } else if (value.equals(REMOVE_INDICATOR)) {
        return new AmendmentPatchRemoval(path, null);
    }
    return new AmendmentPatch(AmendmentPatchOperation.REPLACE, path, AmendmentValue.from(value));
}
````

Reply:
ok, it is used now :)